modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -986,6 +986,165 @@ public void testDeleteObjectEntryAccountRestriction() throws Exception {
+					).build();
+				}
+			},
+			ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		ObjectEntry objectEntry2 = _objectEntryManager.addObjectEntry(
+			_simpleDTOConverterContext, objectDefinition2,
+			new ObjectEntry() {
+				{
+					properties = HashMapBuilder.<String, Object>put(
+						"r_oneToManyRelationship_" +
+							objectDefinition1.getPKObjectFieldName(),
+						objectEntry1.getId()
+					).build();
+				}
+			},
+			ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		Role role = _roleLocalService.addRole(
+			_adminUser.getUserId(), Role.class.getName(), 0,
+			RandomTestUtil.randomString(),
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			Collections.emptyMap(), RoleConstants.TYPE_REGULAR,
+			StringPool.BLANK, null);
+
+		_resourcePermissionLocalService.setResourcePermissions(
+			_companyId, objectDefinition1.getClassName(),
+			ResourceConstants.SCOPE_COMPANY, String.valueOf(_companyId),
+			role.getRoleId(),
+			new String[] {
+				ActionKeys.DELETE, ActionKeys.PERMISSIONS, ActionKeys.UPDATE,
+				ActionKeys.VIEW
+			});
+
+		_user = _addUser();
+
+		_userLocalService.addRoleUser(role.getRoleId(), _user);
+
+		try {
+			_objectEntryManager.deleteObjectEntry(
+				objectEntry1.getExternalReferenceCode(), _companyId,
+				objectDefinition1, null);
+
+			Assert.fail();
+		}
+		catch (PortalException portalException) {
+			Assert.assertEquals(
+				StringBundler.concat(
+					"Object relationship ",
+					objectRelationship.getObjectRelationshipId(),
+					" does not allow deletes"),
+				portalException.getMessage());
+		}
+
+		objectRelationship =
+			_objectRelationshipLocalService.updateObjectRelationship(
+				objectRelationship.getObjectRelationshipId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
+				objectRelationship.getLabelMap());
+
+		try {
+			_objectEntryManager.deleteObjectEntry(
+				objectEntry1.getExternalReferenceCode(), _companyId,
+				objectDefinition1, null);
+
+			Assert.fail();
+		}
+		catch (ObjectRelationshipDeletionTypeException
+					objectRelationshipDeletionTypeException) {
+
+			String message =
+				objectRelationshipDeletionTypeException.getMessage();
+
+			Assert.assertTrue(
+				message.contains(
+					"User " + _user.getUserId() +
+						" must have DELETE permission for"));
+		}
+
+		objectRelationship =
+			_objectRelationshipLocalService.updateObjectRelationship(
+				objectRelationship.getObjectRelationshipId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_DISASSOCIATE,
+				objectRelationship.getLabelMap());
+
+		_objectEntryManager.deleteObjectEntry(
+			objectEntry1.getExternalReferenceCode(), _companyId,
+			objectDefinition1, null);
+
+		Assert.assertNull(
+			_objectEntryManager.fetchObjectEntry(
+				_simpleDTOConverterContext, objectDefinition1,
+				objectEntry1.getId()));
+
+		PermissionThreadLocal.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(_adminUser));
+
+		PrincipalThreadLocal.setName(_adminUser.getUserId());
```

Comment:
@murilostodolni Ideally we should keep the original state, like:

```
String originalName = PrincipalThreadLocal.getName();
PermissionChecker originalPermissionChecker =
	PermissionThreadLocal.getPermissionChecker();

try {
	PrincipalThreadLocal.setName(_adminUser.getUserId());
	PermissionThreadLocal.setPermissionChecker(
		PermissionCheckerFactoryUtil.create(_adminUser));

    ....
}
finally {
	PrincipalThreadLocal.setName(originalName);
	PermissionThreadLocal.setPermissionChecker(
		originalPermissionChecker);
}
```

Reply:
Perfect! Thank you